zng/union.go

```diff
@@ -0,0 +1,87 @@
+package zng
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/mccanne/zq/zcode"
+)
+
+type TypeUnion struct {
+	id    int
+	Types []Type
+}
+
+func NewTypeUnion(id int, types []Type) *TypeUnion {
+	return &TypeUnion{id, types}
+}
+
+func (t *TypeUnion) ID() int {
+	return t.id
+}
+
+func (t *TypeUnion) SetID(id int) {
+	t.id = id
+}
+
+func (t *TypeUnion) TypeIndex(index int) (Type, error) {
+	if index < 0 || index+1 > len(t.Types) {
```

Comment:
Nit: I think `index >= len(t.Types)` would be clearer.

Reply:
you bet. "pourquoi faire simple quand on peut faire compliqué" :)